akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala

```diff
@@ -213,6 +185,8 @@ private[remote] abstract class InboundCompression[T >: Null](
       finally scheduleNextTableAdvertisement()
   }
 
+  private[this] var advertisementInProgress = false
```

Comment:
should be volatile or atomic actually, since it may be triggered from different threads.

I wanted to make InboundCompression extend AtomicReference similar like OutboundCompression does already, then this state and the tables would be in the State object.